node_modules/gh-orgunit/lib/internal/dao.js

```diff
@@ -415,3 +416,377 @@ var getOrgUnitCalendar = module.exports.getOrgUnitCalendar = function(orgUnit, s
+ */
+var _exportOrgUnitCSV = function(rootId, orgUnits, callback) {
+    // Get the simplified tree by exporting it to JSON first
+    _exportOrgUnitJSON(rootId, orgUnits, function(err, exportedOrgUnit) {
+        if (err) {
+            return callback(err);
+        }
+
+        var columns = _getExportCSVColumns(exportedOrgUnit);
+        if (!columns) {
+            return callback({'code': 400, 'msg': 'Tried to export an organisational unit that has children with the same type of different levels'});
+        }
+
+        var records = _getExportCSVRecords(exportedOrgUnit);
+
+        // Stringify the records into CSV
+        var options = {
+            'header': true,
+            'columns': columns
+        };
+        return csv.stringify(records, options, callback);
+    });
+};
+
+/**
+ * Get the column headers for a CSV export of an organisational unit tree. If the tree contained
+ * the same type of organisational unit on different levels, this function will return `null`.
+ * This indicates that the specific tree is malformed and cannot be exported to CSV
+ *
+ * @param  {Object}         exportedOrgUnit     A tree as generated by `exportOrgUnitJSON`
+ * @return {String[]}                           The column headers for a CSV export
+ * @api private
+ */
+var _getExportCSVColumns = function(exportedOrgUnit) {
+    // Walk the tree and determine the type of each nodes. We retain this information so we
+    // can collect the correct column headers for the CSV file
+    var types = {};
+    var malformed = false;
+    var visitNodeForType = function(node, level) {
+        var type = node.type;
+        if (types[type] && types[type] !== level) {
+            log().warn({'orgUnit': rootId}, 'Tried to export an organisational unit that has children with the same type of different levels');
+            malformed = true;
+        }
+
+        types[type] = level;
+
+        // Visit the node's children
+        _.each(node.children, function(child) {
+            visitNodeForType(child, level + 1);
+        });
+    };
+
+    // Visit each node, starting with the root
+    visitNodeForType(exportedOrgUnit, 0);
+
+    // We can't support trees with the same type on different levels
+    if (malformed) {
+        return null;
+    }
+
+    var columns = _.chain(types)
+        .pairs()
+        // Sort the types ascending on their level
+        .sortBy(function(item) {
+            return item[1];
+        })
+        // Return the columns for each organisational unit type
+        .map(function(item) {
+            return [
+                item[0] + '_id',
+                item[0] + '_externalId',
+                item[0] + '_displayName',
+                item[0] + '_description',
+                item[0] + '_published'
+            ];
+        })
+        .value();
+
+    // Add the column headers for series, events and organisers
+    columns.push(['series_id', 'series_externalId', 'series_displayName', 'series_description']);
```

Comment:
You can just do `columns.push('series_id', 'series_externalId', 'series_displayName', 'series_description');` so you don't have to flatten afterwards